### PR TITLE
Gateways are now organized

### LINKED ISSUE #486: Gateway and CoGateway should be Organized.
- Gateway and CoGateway should be organised contracts.
- Remove the organisation address from the following contract constructors.
   - GatewayBase
   - Gateway
   - CoGateway 
   - EIP20Gateway
   - EIP20CoGateway
- Remove the `organisation` variable from GatewayBase.
- Write/Update the `constructors.js` tests for all the above contracts and also the modified functions.

### PR DESCRIPTION
**Pull request by schemar**
⚠️ First review and merge #514 ⚠️

I made the `GatewayBase` organized. As a result, the contracts that
directly or indirectly extend the `GatewayBase` must now provide an
address of a worker manager contract that implements the interface
`IsWorkerInterface`. All relevant modifiers were replaced by
`onlyWorker`.

Tests were updated to reflect the change. In order to easily test
organized contracts, I created a `MockWorkerManager` that can be used in
tests and has a single valid worker address.

I also re-organized the `contracts/test` directory a little bit with
additional sub-directories.

Fixes #486

### CODE REVIEW

File: contracts/gateway/EIP20CoGateway.sol [hunk 1]
```diff
@@ -202,23 +203,23 @@ contract EIP20CoGateway is GatewayBase {
      * @param _utilityToken The utility token address that will be used for
      *                      minting the utility token.
      * @param _core Core contract address.
-     * @param _bounty The amount that facilitator will stake to initiate the
+     * @param _bounty The amount that facilitator will stakes to initiate the
```
Comment (deepesh-kn) [inline contracts/gateway/EIP20CoGateway.sol]: 😳 
`The amount that facilitator stakes to initiate the stake process`

File: contracts/test/gateway/MockEIP20Gateway.sol [hunk 2]
```diff
@@ -0,0 +1,24 @@
+pragma solidity ^0.5.0;
+
+import "../../gateway/EIP20Gateway.sol";
+import "../../lib/IsMemberInterface.sol";
+
+contract MockEIP20Gateway is EIP20Gateway {
```
Comment (deepesh-kn) [inline contracts/test/gateway/MockEIP20Gateway.sol]: I think after removing gateway linking this is not needed anymore.

File: contracts/gateway/EIP20CoGateway.sol (hunk 1 above)
Comment (schemar) [reply contracts/gateway/EIP20CoGateway.sol]: 👍

File: contracts/test/gateway/MockEIP20Gateway.sol (hunk 2 above)
Comment (schemar) [reply contracts/test/gateway/MockEIP20Gateway.sol]: Wow good catch 🐙

**Pull request merged**